tools/reviewer/webapp/src/app/dashboard/diff/diff-discussion/diff-discussion.component.ts

```diff
@@ -1,36 +1,103 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnChanges, OnInit } from '@angular/core';
 
-import { Comment, Thread } from '@/shared/proto';
-import { AuthService } from '@/shared/services';
+import { Comment, Diff, Thread } from '@/shared/proto';
+import { AuthService, FirebaseService, NotificationService } from '@/shared/services';
 import { DiffService } from '../diff.service';
 
 // The component implements UI of thread list of the diff
 // How it looks: https://i.imgur.com/cc6XITV.jpg
+
+// There's used some methods and interfaces from file-changes
+// TODO: make code DRY and reuse it from one place
 @Component({
   selector: 'diff-discussion',
   templateUrl: './diff-discussion.component.html',
   styleUrls: ['./diff-discussion.component.scss'],
   providers: [DiffService],
 })
-export class DiffDiscussionComponent {
+export class DiffDiscussionComponent implements OnInit, OnChanges {
   displayedColumns = ['discussions'];
-  @Input() threads: Thread[];
+  allThreads: Thread[];
```

Comment:
what's the difference between `threads` and `allThreads`

Reply:
Yes yes, we're discussing it right now:
https://appstory.slack.com/archives/C8ZHYNGRZ/p1536759146000100

short answer: allThreads = thread + diffThreads
https://github.com/google/startup-os/blob/master/tools/reviewer/local_server/service/code_review.proto#L84